docs/source/tuning.rst

```diff
@@ -0,0 +1,139 @@
+The broken Submap is broken forever though.
+
+TODO(hrapp): VIDEO
+
+Since the problem here is slippage inside a submap, it is a local SLAM issue.
+So let's turn off global SLAM to not mess with our tuning.
+
+.. code-block:: lua
+
+   SPARSE_POSE_GRAPH.optimize_every_n_scans = 0
+
+Corect size of submaps
+^^^^^^^^^^^^^^^^^^^^^^
+
+Local SLAM drifts over time, only loop closure can fix this drift.
+Submaps must be small enough so that the drift inside them is below the resolution, so that they are locally correct.
+On the other side, they should be so large as being distinct for loop closure to work properly.
+The size of submaps is configured through ``TRAJECTORY_BUILDER_2D.submaps.num_range_data``.
+Looking at the individual submaps for this example they already fit the two constraints rather well, so we assume this parameter is well tuned.
+
+The choice of scan matchers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The idea behind local SLAM is to use sensor data of other sensors besides the range finder to predict where the next scan should be inserted into the submap.
+Then, the ``CeresScanMatcher`` takes this as prior and finds the best spot where the scan match fits the submap.
+It does this by interpolating the submap and sub-pixel aligning the scan.
+This is fast, but cannot fix errors that are significantly larger than the resolution of the submaps.
+If your sensor setup and timing is reasonable, using only the ``CeresScanMatcher`` is usually the best choice to make.
+
+If you do not have other sensors or you do not trust them, Cartographer also provides a ``RealTimeCorrelativeScanMatcher``.
+It uses an approach similar to how scans are matched against submaps in loop closure, but instead it matches against the current submap.
+The best match is then used as prior for the ``CeresScanMatcher``.
+This scan matcher is very expensive and will essentially override any signal from other sensors but the range finder, but it is robust in feature rich environment.
+
+Tuning the correlative scan matcher
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+TODO
+
+Tuning the ``CeresScanMatcher``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In our case, the scan matcher can freely move the match forward and backwards without impacting the score.
+We'd like to penalize this situation by making the scan matcher pay more for deviating from the prior that it got.
+The two parameters controlling this are ``TRAJECTORY_BUILDER_2D.ceres_scan_matcher.translation_weight`` and ``rotation_weight``.
+The higher, the more expensive it is to move the result away from the prior, or in other words: scan matching has to generate a higher score in another position to be accepted.
+
+For instructional purposes, let's make deviating from the prior really expensive:
+
+.. code-block:: lua
+
+   TRAJECTORY_BUILDER_2D.ceres_scan_matcher.translation_weight = 1e3
+
+TODO: video
+
+This allows the optimizer to pretty liberally overwrite the scan matcher results, and the pose floats freely.
```

Comment:
This sentence isn't clear to me. I'd expect the setting above to do very little optimisation, but instead just return something very close to the prior - is that correct? I also don't know what "the pose floats freely" means.

Again, ignore if the video makes it clear.